### PR TITLE
Add LinkCollection#structured_data

### LINKED ISSUE #25: JSON-LD support
I've started using this gem for breadcrumbs on my site, and it's been great so far. I initially used the built-in generator for the breadcrumb HTML, but it didn't quite fit my requirements so I opted for creating them by hand using the `breadcrumbs` method. However in doing so, I lost the semantic breadcrumb functionality. I'm working on a replacement using the JSON-LD/`ld+json` version (as seen [here](https://developers.google.com/search/docs/data-types/breadcrumb#json-ld)). It would be awesome if this library supported that natively.

I'll likely have a PR to add this functionality before long. I hope you'll be willing to merge it. Stay tuned.

### PR DESCRIPTION
**Pull request by dkniffin**
Fixes #25

@kzkn I can definitely add tests to this, but I wanted to see what you think of it first.

### CODE REVIEW

**Comment by kzkn** [comment]
@dkniffin 
LGTM. Keep going 😆
 
It would be great if you could include a sample of what you can write in the README.

**Comment by dkniffin** [comment]
@kzkn I've updated the previous commit to include the test and the docs update. Let me know if you need anything else before merging this.

File: lib/gretel/renderer.rb [hunk 1]
```diff
@@ -159,6 +159,27 @@ def initialize(context, links, options = {})
...
 
+      # Returns a hash matching the JSON-LD Structured Data schema
+      # https://developers.google.com/search/docs/data-types/breadcrumb#json-ld
+      def structured_data(url_base:)
+        url_base = url_base.chomp("/") # Remove trailing / if present
+
+        items = @links.each_with_index.map do |link, i|
+          {
+            "@type": "ListItem",
+            "position": i+1,
```
Comment (kzkn) [inline lib/gretel/renderer.rb:170]:
```suggestion
            "position": i + 1,
```

File: spec/lib/gretel/view_helpers_spec.rb [hunk 2]
```diff
@@ -489,6 +489,40 @@
...
+            "@type": "ListItem",
+            "position": 3,
+            "name": "Contact",
+            "item": "https://example.com/about/contact"
+          }
+        ]
+      }
+
+      expect(breadcrumbs.structured_data(url_base: "https://example.com/")).to eq(expected_result)
+    end
```
Comment (kzkn) [inline spec/lib/gretel/view_helpers_spec.rb:523]: Please add a spec for passing a URL that have no `/` suffix.

**Review by kzkn — CHANGES_REQUESTED** [review]
Almost, LGTM :+1: 
I'm commenting on some of the finer points. Please check.

**Comment by dkniffin** [comment]
@kzkn I've made those two updates you requested

**Review by kzkn — APPROVED** [review]

**Comment by kzkn** [comment]
@dkniffin thanks!

**Comment by dkniffin** [comment]
@kzkn Awesome, thank you! One more thing: can you bump the version so I can use the gem without pointing my Gemfile to github?

**Comment by kzkn** [comment]
@dkniffin released as 4.2.0 :tada:

**Pull request merged**